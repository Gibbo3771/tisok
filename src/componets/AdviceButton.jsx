import React from "react";

const AdviceButton = props => {
  return (
    <a className="advice-get" onClick={props.onRequestRandom}>
      Advice
    </a>
  );
};

export default AdviceButton;
