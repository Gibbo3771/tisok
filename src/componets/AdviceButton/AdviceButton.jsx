import React from "react";

export default class AdviceButton extends React.Component {
  onClick = () => {
    this.props.onClick();
    this.props.history.push("/");
  };

  render() {
    return (
      <button className="advice-get" onClick={this.onClick}>
        (｡^‿^｡)
      </button>
    );
  }
}
