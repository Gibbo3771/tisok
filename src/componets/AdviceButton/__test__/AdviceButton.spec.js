// @flow
import React from "react";
import AdviceButton from "../AdviceButton";

const props = {
  onClick: sinon.spy(),
  history: {
    push: sinon.spy()
  }
};
describe("AdviceButton", () => {
  beforeEach(() => {
    props.onClick.resetHistory();
    props.history.push.resetHistory();
  });
  it("renders correctly", () => {
    const tree = shallow(
      <AdviceButton history={props.history} onClick={props.onClick} />
    );
    expect(tree).toMatchSnapshot();
  });

  it("pushes '/' to the browser history", () => {
    const wrapper = shallow(
      <AdviceButton history={props.history} onClick={props.onClick} />
    );
    wrapper.find("button").simulate("click");
    expect(props.history.push.calledOnceWith("/")).toBeTruthy();
  });

  it("responds to click events", () => {
    const wrapper = shallow(
      <AdviceButton history={props.history} onClick={props.onClick} />
    );
    expect(props.onClick.calledOnce).toBeFalsy();
    wrapper.find("button").simulate("click");
    expect(props.onClick.calledOnce).toBeTruthy();
  });
});
