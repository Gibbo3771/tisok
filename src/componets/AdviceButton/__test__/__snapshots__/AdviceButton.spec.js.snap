// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdviceButton renders correctly 1`] = `ShallowWrapper {}`;
